intellij/ast/src/main/kotlin/motif/ast/intellij/IntelliJAnnotation.kt

```diff
@@ -43,18 +46,14 @@ class IntelliJAnnotation(
     }
 
     override val type: IrType? by lazy {
-        val nameReferenceElement = psiAnnotation.nameReferenceElement
-        val annotationClass: PsiClass = nameReferenceElement?.resolve() as? PsiClass ?: return@lazy null
+        val annotationClass = annotationClass ?: return@lazy null
         val psiClassType: PsiClassType = PsiElementFactory.SERVICE.getInstance(project).createType(annotationClass)
         IntelliJType(project, psiClassType)
     }
 
     override val members: List<IrMethod> by lazy {
-        psiAnnotation.parameterList.attributes
-                .map {
-                    it.reference!!.resolve() as PsiMethod
-                }
-                .map { IntelliJMethod(project, it, PsiSubstitutor.EMPTY) }
+        val annotationClass = annotationClass ?: return@lazy emptyList<IrMethod>()
+        annotationClass.methods.map { IntelliJMethod(project, it, PsiSubstitutor.EMPTY) }
```

Comment:
possible NPE if annotationClass resolves to null after the elvis operator?

Reply:
This wouldn't compile if that was the case. The `?: return@lazy emptyList<IrMethod>()` is a quick return if `annotationClass` is null.

Reply:
The second reference to `annotationClass` is a reference to the local `val`, not the field.

Reply:
👍